### PR TITLE
bind: split out dnsutils & host binaries

### PR DESCRIPTION
**Pull request by fpletz**
###### Motivation for this change

These tools are commonly used but don't require the other bind binaries. Bind's libs are used, so they've also been split into an extra output.

The old version of host isn't maintained anymore and was removed from Debian back in 2009: https://packages.qa.debian.org/h/host.html
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by NeQuissimus — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -2062,7 +2062,7 @@ in
 
   horst = callPackage ../tools/networking/horst { };
 
-  host = callPackage ../tools/networking/host { };
+  host = bind.host;
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix]: Is a separate host package really required or could this package be deprecated in favor of dnsutils?

Comment (fpletz) [inline pkgs/top-level/all-packages.nix]: `host` was historically and is still a separate package in most Linux distributions. I chose to go with the Debian naming where bind9 also provides the dnsutils and host binary packages: https://packages.qa.debian.org/b/bind9.html. Just noticed, though, that dnsutils actually depends on host.

Not sure about this, we can do it either way in my opinion.

Comment (Mic92) [inline pkgs/top-level/all-packages.nix]: My user expectation would be at least, that I get `host`, when pulling `dnsutils` into systemPackages. How about an additional symlink?

Comment (fpletz) [inline pkgs/top-level/all-packages.nix]: Added a symlink to host in dnsutils.

**Pull request merged**